### PR TITLE
Updated carousel animations

### PR DESCRIPTION
**Pull request by krikof**
# Description

This is a new attempt at creating an animated carousel. Currently it has some flaws, but can be addressed in the future. There are new cards related to these issues. (See Github Project)

This is currently WIP and needs bigger implementations.  As it stands we now have a better looking Idle Carousel. The idea is that when this screen is shown, it cycles through apps and display some information.

The current functionally goes as follows:

The screen we cycle through a list of apps. When the app icon is «in view», we expand the icon into a card, show some information.

### Related Issues

Current issue: [issue](https://github.com/shortcut/appshowcase-android/issues/5)

### Related Figma links

The Figma design for this implementation:

[Figma](https://www.figma.com/file/qVBWfqyKSKrtLg3cN6XDId/Shortcut-App-gallery?node-id=154%3A3408&t=CL2Lk5Pd1HG8peu9-1)

### Related screenshots

# How has this been tested?

Currently this has only been tested on hands. No written tests.
As it stands, I disabled a lot of the features due to lack of implementation. This is work in progress. 

# Checklist

[✅] I have performed a self-review of my own code
[✅ ] I have run my changes with the release build type and verified that there are no obvious ProGuard issues
[❌] I have tested my feature with talkback and text scaling <- Being worked on, will create a ticket.

### Notes

As it currently stands, it doesn't fit the original acceptance criteria of the issue. It should however be merged, as it is a WIP.

### CODE REVIEW

**Comment by arunvijayanpillai** [comment]
Can you remove the [build folder](https://github.com/shortcut/appshowcase-android/tree/updated_carousel/welcome/build) from GitHub? they are auto-generated and we won't be needing them in version control.

**Comment by krikof** [comment]
Will do, you are right!

**Comment by krikof** [comment]
@arunvijayanpillai 

I think I have removed the unneccesary build folder from the project. Great catch Arun, I didn't see that. All should be good as of now (hopefully).

**Comment by arunvijayanpillai** [comment]
> Will do, you are right!

This lot better now. :-) btw update the .gitignore file and then we can be sure this won't get added again.

File: app/src/main/java/io/shortcut/showcase/presentation/idle/view/CarouselScreen.kt [hunk 1]
```diff
@@ -40,67 +41,90 @@ import io.shortcut.showcase.ui.theme.ExtendedShowcaseTheme
...
-    ExtendedShowcaseTheme {
-        IdleBackground()
-    }
-}
-
-@Composable
-private fun IdleContent(
+private fun CarouselContent(
     modifier: Modifier = Modifier,
+    childModifier: Modifier = Modifier,
```
Comment (arunvijayanpillai) [inline app/src/main/java/io/shortcut/showcase/presentation/idle/view/CarouselScreen.kt:65]: I am curious why we needed a child modifier here

File: app/src/main/java/io/shortcut/showcase/presentation/idle/view/CarouselScreen.kt [hunk 2]
```diff
@@ -40,67 +41,90 @@ import io.shortcut.showcase.ui.theme.ExtendedShowcaseTheme
...
     val systemUiController: SystemUiController = rememberSystemUiController()
     systemUiController.setSystemBarsColor(color = ShowcaseThemeCustom.colors.ShowcaseBackground)
     systemUiController.isNavigationBarVisible = false
-
-    IdleContent(
-        apps = carouselAppList,
-        onBackClick = onBackClick
+    CarouselContent(
+        modifier = Modifier.clickable { onScreenClick() },
+        apps = carouselAppList
```
Comment (arunvijayanpillai) [inline app/src/main/java/io/shortcut/showcase/presentation/idle/view/CarouselScreen.kt:57]: Maybe we need a viewmodel here as this data `carouselAppList` shouldn't be accessed here(?)

**Comment by arunvijayanpillai** [comment]
BTW while testing it felt like the collapsing is too quick, especially when the white color turns dark.

**Pull request closed without merging**